api/dp-openapi.yaml

```diff
@@ -784,6 +1132,163 @@ components:
         state:
           type: string
 
+    wordcheck_data:
+      type: object
+      properties:
+        langsWithDict:
+          type: array
+          items:
+            type: string
+        messages:
+          type: object
+        puncChars:
+          type: string
+
+    proof_page:
+      type: object
+      properties:
+        pagename:
+          type: string
+        image_url:
+          type: string
+        text:
+          type: string
+
+    page_text:
+      type: object
+      properties:
+        text:
+          type: string
+
+    round_data:
+      type: object
+      properties:
+        id:
+          type: string
+          example: F1
+        name:
+          type: string
+          description: the translated description
+          example: Formatting Round 1
+        formatting:
+          type: boolean
+          description: true if it is a formatting round
+          example: true
+        document:
+          type: string
+          description: url of the guidelines for the round
+          example: https://www.pgdp.net/wiki/DP_Official_Documentation:Formatting/Formatting_Guidelines
+
+    page_check:
+      description: check text using given languages and ad hoc good words
+      type: object
+      properties:
+        text:
+          type: string
+        auxlanguage:
+          type: string
+        adhocgoodwords:
+          type: array
+          items:
+            type: string
+
+    page_save:
```

Comment:
not sure how swagger would represent this, but should we have 1 type with optional entries for text,  and reason instead of 3 and the oneOf? Not sure if it is even better or not.